### PR TITLE
Test initializing from config_dict via hypothesis

### PR DESCRIPTION
**Pull request by eivindjahren**
Add regression tests for issues described in #2536

## Pre review checklist

- [x] Added appropriate release note label
- [x] PR title captures the intent of the changes, and is fitting for release notes.
- [x] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).

Adding labels helps the maintainers when writing release notes. This is the [list of release note labels](https://github.com/equinor/ert/labels?q=release-notes).

### CODE REVIEW

File: src/ert/_c_wrappers/enkf/ecl_config.py [hunk 1]
```diff
@@ -193,6 +193,16 @@ def num_cpu(self):
         """
         return self._get_num_cpu()
 
+    def __repr__(self):
+        return (
+            "EclConfig(config_dict={"
+            f"'{ConfigKeys.DATA_FILE}: {self.getDataFile()}, "
+            f"'{ConfigKeys.GRID}: {self.get_gridfile()}, "
+            f"'{ConfigKeys.REFCASE}: {self.getRefcaseName()}, "
+            f"'{ConfigKeys.END_DATE}: {self.getEndDate()}, "
```
Comment (valentin-krasontovitsch) [inline src/ert/_c_wrappers/enkf/ecl_config.py:202]: forgot to close single quote?

File: src/ert/_c_wrappers/enkf/analysis_config.py [hunk 2]
```diff
@@ -264,6 +264,23 @@ def haveEnoughRealisations(self, realizations) -> bool:
     def __ne__(self, other):
         return not self == other
 
+    def __repr__(self):
+        return (
+            "AnalysisConfig(config_dict={"
+            f"'UPDATE_LOG_PATH'={realpath(self.get_log_path())}, "
```
Comment (valentin-krasontovitsch) [inline src/ert/_c_wrappers/enkf/analysis_config.py:270]: should the equal sign be a colon with a space, like for the rest of the options?

File: src/ert/_c_wrappers/enkf/queue_config.py [hunk 3]
```diff
@@ -161,6 +161,16 @@ def lsf_server(self):
...
         return self._get_num_cpu()
 
+    def __repr__(self):
+        return (
+            "QueueConfig(config_dict={"
+            f"{ConfigKeys.JOB_SCRIPT}: {self.job_script}, "
+            f"{ConfigKeys.MAX_SUBMIT}: {self.max_submit}, "
+            f"{ConfigKeys.NUM_CPU}: {self.num_cpu}, "
+            f"{ConfigKeys.QUEUE_SYSTEM}: {self.queue_system}, "
+            "})"
```
Comment (valentin-krasontovitsch) [inline src/ert/_c_wrappers/enkf/queue_config.py:171]: no single quotes in this representation?

**Review by valentin-krasontovitsch — CHANGES_REQUESTED** [review]
just a few format things for now

File: src/ert/_c_wrappers/enkf/analysis_config.py (hunk 2 above)
Comment (eivindjahren) [reply src/ert/_c_wrappers/enkf/analysis_config.py]: Fixed

File: src/ert/_c_wrappers/enkf/queue_config.py (hunk 3 above)
Comment (eivindjahren) [reply src/ert/_c_wrappers/enkf/queue_config.py:171]: fixed

File: src/ert/_c_wrappers/enkf/ert_workflow_list.py [hunk 4]
```diff
@@ -162,6 +162,17 @@ def free(self):
...
         return not (self == other)
 
+    def __repr__(self):
+        job_dicts = [
+            {ConfigKeys.NAME: name, ConfigKeys.PATH: self.getJob(name).executable()}
+            for name in self.getJobNames()
+        ]
+        return (
+            "ErtWorkflowList(config_dict={"
+            f"{ConfigKeys.LOAD_WORKFLOW_JOB}: {job_dicts}, "
```
Comment (valentin-krasontovitsch) [inline src/ert/_c_wrappers/enkf/ert_workflow_list.py:172]: should this one also have single quotes? hope i'm not bugging you too much about format stuff...

**Comment by valentin-krasontovitsch** [comment]
haven't managed to read through all of it yet, but ooohhhh, this is very yummy! nice to see some of the logic for the config represented in the tests, and hypothesis looks very handy, had not seen it before.

thank you!

File: src/ert/_c_wrappers/enkf/ert_workflow_list.py (hunk 4 above)
Comment (eivindjahren) [reply src/ert/_c_wrappers/enkf/ert_workflow_list.py]: No problem! Fixed

**Review by valentin-krasontovitsch — APPROVED** [review]
OK I'm not gonna try to read and understand the tests in detail, as that would probably take a long time and I trust your work. running them works, and I like how you have linked to the issues for tests that are skipped / failing. if there are problems, i suppose we will find them once we start working on the config. thank you!

**Comment by eivindjahren** [comment]
> OK I'm not gonna try to read and understand the tests in detail, as that would probably take a long time and I trust your work. running them works, and I like how you have linked to the issues for tests that are skipped / failing. if there are problems, i suppose we will find them once we start working on the config. thank you!

Yes, I think its safe to merge this. The tests run fast enough, and are not flaky so it won't bother anyone. Also, it will not introduce bugs. The idea is to use these as acceptance tests for our work going forward.

**Pull request merged**